app/client/src/PluginActionEditor/components/PluginActionResponse/components/Response/utils/getErrorFromActionResponse.ts

```
@@ -23,6 +23,10 @@ export const getErrorMessageFromActionResponse = (
     );
   }
 
-  // Return body if plugin type is DB, otherwise return default error message
-  return pluginType === PluginType.DB ? body : DEFAULT_ERROR_MESSAGE;
+  // Return body if plugin type is DB/SAAS/ExternalSAAS, otherwise return default error message
+  return pluginType === PluginType.DB ||
+    pluginType === PluginType.SAAS ||
```

Does this change takes care of the error for Saas Plugin type ?

---

yes saas plugin also had issues, it was masking the error message, fixed it for that as well